common/environment/environment.hpp

```diff
@@ -175,7 +175,7 @@ interface IEnvironmentFactory : extends IInterface
 
 
 class StringBuffer;
-extern "C" ENVIRONMENT_API IEnvironmentFactory * getEnvironmentFactory();
+extern "C" ENVIRONMENT_API IEnvironmentFactory * getEnvironmentFactory(bool update = false);
```

Comment:
Would it be cleaner for this parameter to not have a default parameter - there are several cases which don't pass any parameter.  When should true be passed, and when should false?

Reply:
Yes probably, always debatable on default params like this.